File: frontend/src/types/feedback.ts
export enum MatchQualityEnum {
  VERY_GOOD = "très bon",
  GOOD = "bon",
  MEDIUM = "moyen",
  POOR = "faible",
  UNACCEPTABLE = "inacceptable"
}

export interface Feedback {
  id: string;
  match_id: string;
  user_id: string;
  user_type: "recruiter" | "candidate";
  rating: number;
  match_quality: MatchQualityEnum;
  comment?: string;
  algorithm_version: string;
  created_at: string;
  updated_at: string;
}

export interface FeedbackStats {
  average_rating: number;
  total_feedbacks: number;
  quality_distribution: Record<MatchQualityEnum, number>;
  latest_feedback_date: string | null;
}

export interface FeedbackFilters {
  algorithm_version?: string;
  min_rating?: number;
  max_rating?: number;
  start_date?: string;
  end_date?: string;
  user_type?: "recruiter" | "candidate";
}

File: matching-service/schemas/feedback.py
from pydantic import BaseModel, Field, validator, UUID4
from datetime import datetime
from typing import Optional, Literal
from enum import Enum


class MatchQualityEnum(str, Enum):
    VERY_GOOD = "très bon"
    GOOD = "bon"
    MEDIUM = "moyen"
    POOR = "faible"
    UNACCEPTABLE = "inacceptable"


class FeedbackBase(BaseModel):
    rating: int = Field(..., ge=1, le=5, description="Note de 1 à 5")
    match_quality: MatchQualityEnum
    comment: Optional[str] = Field(None, max_length=1000, description="Commentaire optionnel")


class FeedbackCreate(FeedbackBase):
    match_id: UUID4
    
    @validator('comment')
    def clean_comment(cls, v):
        if v:
            return v.strip()
        return v


class FeedbackResponse(FeedbackBase):
    id: UUID4
    match_id: UUID4
    user_id: UUID4
    user_type: Literal["recruiter", "candidate"]
    algorithm_version: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
        json_encoders = {datetime: lambda v: v.isoformat()}


class FeedbackStats(BaseModel):
    average_rating: float
    total_feedbacks: int
    quality_distribution: dict[MatchQualityEnum, int]
    latest_feedback_date: Optional[datetime]